test/MUXControlsTestApp/VisualTreeTestBase.cs

```diff
@@ -0,0 +1,360 @@
+#if USING_TAEF
+           return TestContext.Properties.Contains(key);
+#else
+            return TestContext.Properties.ContainsKey(key);
+#endif
+        }
+    }
+
+    class MasterFileStorage
+    {
+        private StorageFolder _storage;
+        public string StorageLocation { get; private set; }
+        public string ExpectedMasterFileName { get; private set; }
+        public string BestMatchedMasterFileName { get; private set; }
+        public MasterFileStorage(bool useLocalStorage, string masterFileNamePrefix, string theme)
+        {
+            _storage = useLocalStorage ? ApplicationData.Current.LocalFolder : KnownFolders.MusicLibrary;
+            string prefix = String.IsNullOrEmpty(theme) ? masterFileNamePrefix : masterFileNamePrefix + "-" + theme;
+            ExpectedMasterFileName = GetExpectedMasterFileName(prefix);
+            BestMatchedMasterFileName = SearchBestMatchedMasterFileName(prefix);
+            StorageLocation = useLocalStorage ? ApplicationData.Current.LocalFolder.Path : "MusicLibrary";
+        }
+
+        public void LogMasterFile(string fileName, string content)
+        {
+            LogMasterFile(_storage, fileName, content);
+        }
+
+        private string GetExpectedMasterFileName(string fileNamePrefix)
+        {
+            return String.Format("{0}-{1}.xml", fileNamePrefix, PlatformConfiguration.GetCurrentAPIVersion());
+        }
+
+        private string SearchBestMatchedMasterFileName(string fileNamePrefix)
+        {
+            for (ushort version = PlatformConfiguration.GetCurrentAPIVersion(); version >= 2; version--)
+            {
+                string fileName = String.Format("{0}-{1}.xml", fileNamePrefix, version);
+                if (MasterFileStorage.IsMasterFilePresent(fileName))
+                {
+                    return fileName;
+                }
+            }
+            {
+                string fileName = fileNamePrefix + ".xml";
+                if (MasterFileStorage.IsMasterFilePresent(fileName))
+                {
+                    return fileName;
+                }
+            }
+            return null;
+        }
+
+        public static string GetMasterFileContent(string fileName)
+        {
+            return GetMasterFileContentAsync(fileName).Result;
+        }
+
+        public static bool IsMasterFilePresent(string fileName)
+        {
+            try
+            {
+                GetMasterFileContentAsync(fileName).Wait();
+                return true;
+            }
+            catch (AggregateException ae)
+            {
+                ae.Handle((x) =>
+                {
+                    return (x is FileNotFoundException); // This we know how to handle.
+                });
+                return false;
+            }
+        }
+
+        public static void LogMasterFile(StorageFolder storageFolder, string fileName, string content)
+        {
+            LogMasterFileAsync(storageFolder, fileName, content).Wait();
```

Comment:
Again, the method name needs to advertise that you're doing blocking IO, not hide it.

Reply:
It's LogMasterFile and LogMasterFileAsync